peps/pep-0817.rst

```diff
@@ -0,0 +1,2528 @@
+package in question.
+
+A few existing approaches are explored in the subsequent subsections.
+These include maintaining separate package indexes for different
+hardware configurations, bundling all potential variants into a single
+wheel of considerable size, or using separate package names
+(``mypackage-gpu``, ``mypackage-cpu``, etc.). Each of these approaches
+has significant drawbacks.
+
+According to the `2024 Python Developers Survey
+<https://lp.jetbrains.com/python-developers-survey-2024/#purposes-for-using-python>`__,
+a significant portion of respondents over the last years have been
+successively using Python for scientific computing purposes, covering
+such areas as Data analysis (steadily over 40% respondents), Machine
+learning (grown to 40% in 2024) and Data engineering (around 30%).
+Many of these use cases are directly impacted by suboptimal
+packaging.
+
+This issue is often crossing the boundaries of scientific computing.
+Wheels currently cannot express more specific CPU requirements, forcing
+the maintainers to build and ship only for the oldest CPU baseline they
+wish to support. Newer CPU features are sometimes utilized through
+runtime dispatching, but for example this does not let software benefit
+from the performance improvements related to auto-vectorization for
+x86-64-v4 CPUs.
+
+As illustrated by `archspec <https://github.com/archspec/archspec>`_,
+the ability to optimize a package for a specific architecture can lead
+to significant performance improvement.
+
+    .. figure:: pep-0817/avx512_gromacs_benchmark.svg
+        :alt: A bar graph comparing GROMACS performance (in ns/day) with
+              various targets. The first two bars are labeled "yum
+              (2018.8)" and "generic (SSE2)", reach about 1.0 ns/day and
+              are both marked as "SSE2". The next bar is labeled
+              "ivybridge" ("AVX") and reaches almost 1.5 ns/day. Two
+              following bars are labeled "haswell" and "broadwell" (both
+              "AVX2") and exceed 1.5 ns/day slightly. The last two bars
+              are labeled "skylake_avx512" and "cascadelake" (both
+              "AVX512") and reach almost 2.0 ns/day.
+
+        Performance of GROMACS 2020.1 built for different generations of
+        CPUs. Vertical axis shows performance expressed in ns/day, a
+        GROMACS-specific measure of simulation speed (higher is better).
+
+    Compiling `GROMACS <https://www.gromacs.org/>`_ for architectures
+    that can exploit the AVX-512 instructions supported by the Intel
+    Cascade Lake microarchitecture gives an additional 18% performance
+    improvement relative to using AVX2 instructions, with a speedup of
+    about 70% compared to a generic GROMACS installation with only SSE2.
+
+    — `archspec: A library for detecting, labeling, and reasoning about
+    microarchitectures
+    <https://tgamblin.github.io/pubs/archspec-canopie-hpc-2020.pdf>`__
```

Comment:
Would suggest making this entire part it's own subsection, moving to the appendix, or deleting. 

I think the general argument that packaging can't target specific hardware beyond the most generic CPU architecture is already compelling, this text doesn't add much. That said, the general argument could be made more clearly and in a distilled form.